bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -500,3 +527,16 @@ func NewBootstrapTokenString(token string) (*BootstrapTokenString, error) {
 
 	return &BootstrapTokenString{ID: substrs[1], Secret: substrs[2]}, nil
 }
+
+// Patches contains options related to applying patches to components deployed by kubeadm.
+type Patches struct {
+	// Directory is a path to a directory that contains files named "target[suffix][+patchtype].extension".
+	// For example, "kube-apiserver0+merge.yaml" or just "etcd.json". "target" can be one of
+	// "kube-apiserver", "kube-controller-manager", "kube-scheduler", "etcd". "patchtype" can be one
+	// of "strategic" "merge" or "json" and they match the patch formats supported by kubectl.
+	// The default "patchtype" is "strategic". "extension" must be either "json" or "yaml".
+	// "suffix" is an optional string that can be used to determine which patches are applied
+	// first alpha-numerically.
```

Comment:
We should document how the user is expected to provide patches in the target directory (I assume via extra files specified in the KubeadmConfig)